cosine_similarity_model.py

```diff
@@ -0,0 +1,179 @@
+import json
+import numpy as np
+from sklearn.metrics.pairwise import cosine_similarity
+from pprint import pprint
+from collections import Counter
+import logging
+
+def setup_logger():
+    logger = logging.getLogger('logfile.log')
+    logger.setLevel(logging.INFO)
+    handler = logging.StreamHandler()
+    formatter = logging.Formatter('%(asctime)s - %(levelname)s - %(message)s')
+    handler.setFormatter(formatter)
+    logger.addHandler(handler)
+    return logger
+
+
+logger = setup_logger()
+
+
+def load_movies_from_json(file_path):
+    with open(file_path, "r", encoding="utf-8") as file:
+        data = json.load(file)
+    return data
+
+
+def extract_genres(movies):
+    all_genre_ids = set()
```

Comment:
repeated manual extraction of data using nested loops is inefficient
you should use pandas for data manipulation, it will simplify much of the code

for example, you can create a DataFrame directly from the MongoDB collection
`movies_df = pd.DataFrame(list(collection.find()))`
then, genres can be extracted easily like this
`genres = movies_df['genres'].apply(lambda x: [i['name'] for i in x])`
use genre names instead of ids